src/main/java/org/generationcp/middleware/service/impl/inventory/LotServiceImpl.java

```diff
@@ -187,11 +209,82 @@ private void updateLots(final List<Lot> lots, final LotMultiUpdateRequestDto lot
 				}
 
 				this.daoFactory.getLotDao().saveOrUpdate(lot);
+
+				if (!CollectionUtils.isEmpty(lotUpdateDto.getAttributes())) {
+					lotUpdateDto.getAttributes().forEach((key, value) -> {
+						final String variableNameOrAlias = key.toUpperCase();
+						this.saveOrUpdateAttribute(attributeVariablesNameMap, attributesMap, lot,
+							variableNameOrAlias, value, conflictErrors);
+					});
+				}
 			}
 		} catch (final Exception e) {
 			LOG.error(e.getMessage(), e);
 			throw new MiddlewareRequestException("", "common.middleware.error.update.lots");
 		}
+
+		if (!conflictErrors.isEmpty()) {
+			throw new MiddlewareRequestException("", conflictErrors);
+		}
+	}
+
+	private void saveOrUpdateAttribute(final Map<String, Variable> attributeVariables,
+		final Map<Integer, List<LotAttribute>> attributesMap, final Lot lot,
+		final String variableNameOrAlias, final String value, final Multimap<String, Object[]> conflictErrors) {
+		// Check first if the code to save is a valid Attribute
+		if (attributeVariables.containsKey(variableNameOrAlias) && StringUtils.isNotEmpty(value)) {
+			final Variable variable = attributeVariables.get(variableNameOrAlias);
+			final List<LotAttribute> lotAttributes = attributesMap.getOrDefault(lot.getId(), new ArrayList<>());
+			final List<LotAttribute> attributesByType =
```

Comment:
@cheryleigh I think we can modify this variable name, this is used to find existing attributes for a given type, right? Something like existingLotAttributesForVariable?